prboom2/src/dsda/hud_components/big_ammo.c

```diff
@@ -42,9 +45,24 @@ static void dsda_DrawComponent(void) {
     return;
 
   ammo = player->ammo[ammo_type];
+  max_ammo = player->maxammo[weaponinfo[player->readyweapon].ammo];
+
+  if (ammo == max_ammo)
```

Comment:
Take a look at `big_health.c` to get an idea of how I would format this section.